tests/python/gaia-ui-tests/gaiatest/gaia_test.py

```diff
@@ -117,8 +117,13 @@ def displayed_app(self):
                        origin=result.get('origin'))
 
     def switch_to_displayed_app(self):
+        from gaiatest.apps.lockscreen.app import LockScreen
+
         self.marionette.switch_to_default_content()
-        self.marionette.switch_to_frame(self.displayed_app.frame)
+        displayed_app = self.displayed_app
+        if displayed_app.origin != LockScreen.origin:
+            # The lockscreen app is not in an iFrame. It's a div in the System app.
+            self.marionette.switch_to_frame(self.displayed_app.frame)
```

Comment:
I'll make the comment more clear in the PR. If we just do `self.marionette.switch_to_frame(self.displayed_app.frame)`, we actually switch to the homescreen, even though the lockscreen is displayed. This is due to the fact that the lockscreen is not an Iframe